oat-sa/extension-tao-itemqti-pic
# Feature/ADF-1418/Migrate sass

### Related to: [ADF-1417](https://oat-sa.atlassian.net/browse/ADF-1417), [ADF-1418](https://oat-sa.atlassian.net/browse/ADF-1418), and [ADF-1483](https://oat-sa.atlassian.net/browse/ADF-1483)

Requires: 

### Summary

Update the stylesheets using proper SASS functions.

BREAKING CHANGE:
- The package now requires Node.js 14.17+
- Replace Node-Sass with Dart-Sass

### Details

To properly support Node 18, the Node-SASS needs to be replaced by Dart-SASS.
This lib is the official version of SASS, and it has breaking changes for the set of supported features (see [SASS: breaking changes](https://sass-lang.com/documentation/breaking-changes/extend-compound)).
Among others:
- compound selectors are no longer supported
- division is not supported outside of `calc`
- color computation must pass through the color functions see https://github.com/sass/sass/issues/2543

### How to test
Pre-requisite:
- please make sure that other extensions are up to date. At least `tao-core` needs to be in version `53.0.0`. 
- you need to run `npm ci`  on both `views` and `views/build`.

Steps for testing:
- install the extension
- compile the styles: `cd tao/views/build && npx grunt sass:qtiitempic`
- add the PIC to an item (edit an item, then activate the student tool)
![image](https://github.com/oat-sa/extension-tao-itemqti-pic/assets/1500098/2e4123b6-ca1b-4e64-bb5b-12b470b1fd63)
- click on the button in the student toolbar and check the right-side panel
![image](https://github.com/oat-sa/extension-tao-itemqti-pic/assets/1500098/4f616b4b-3d30-4613-a4b8-b9bed67fa21f)


[ADF-1417]: https://oat-sa.atlassian.net/browse/ADF-1417?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[ADF-1418]: https://oat-sa.atlassian.net/browse/ADF-1418?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[ADF-1483]: https://oat-sa.atlassian.net/browse/ADF-1483?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
